ethexe/service/src/lib.rs

```diff
@@ -84,15 +84,25 @@ pub enum NetworkMessage {
 
 impl Service {
     pub async fn new(config: &Config) -> Result<Self> {
-        let blob_reader = Arc::new(
-            ethexe_observer::ConsensusLayerBlobReader::new(
-                &config.ethereum.rpc,
-                &config.ethereum.beacon_rpc,
-                config.ethereum.block_time,
+        let mock_blob_reader: Option<Arc<MockBlobReader>> = if config.node.dev {
+            Some(Arc::new(MockBlobReader::new(config.ethereum.block_time)))
+        } else {
+            None
+        };
+
+        let blob_reader: Arc<dyn ethexe_observer::BlobReader> = if config.node.dev {
+            mock_blob_reader.clone().unwrap().clone()
+        } else {
+            Arc::new(
+                ethexe_observer::ConsensusLayerBlobReader::new(
+                    &config.ethereum.rpc,
+                    &config.ethereum.beacon_rpc,
+                    config.ethereum.block_time,
+                )
+                .await
+                .with_context(|| "failed to create blob reader")?,
             )
-            .await
-            .with_context(|| "failed to create blob reader")?,
-        );
+        };
```

Comment:
could be flattened into single if block 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=62a69e942fc810a88c47654c68df0503